RUFAS/EEE/emissions.py

```diff
@@ -283,387 +205,438 @@ def _get_feed_emissions_data(
+        daily_farmgrown_feed_emission_and_resource_by_feed_id: dict[RUFAS_ID, dict[int, dict[str, float]]] = (
+            defaultdict(dict)
+        )
+
+        harvest_dates_by_feed_id: dict[RUFAS_ID, list[int]] = {
+            feed_id: sorted(
+                [
+                    harvest_date
+                    for harvest_date in harvest_yield_by_field[field_name]
+                    if harvest_yield_by_field[field_name][harvest_date]["feed_id"] == feed_id
+                ]
             )
-            crop["potassium_fertilizer_used"] += fertilizer_application["potassium"] * split_factor
-            crop["potassium_fertilizer_embedded_CO2_emissions"] += (
-                fertilizer_application["potassium"] * split_factor * EMBEDDED_POTASSIUM_FERTILIZER_EMISSIONS_FACTOR
+            for field_name in harvest_yield_by_field
+            for feed_id in set(
+                harvest_yield_by_field[field_name][harvest_date]["feed_id"]
+                for harvest_date in sorted(list(harvest_yield_by_field[field_name].keys()))
             )
+        }
 
-    def _filter_results(
-        self, filters: dict[str, Any], date_cutoff: date, year_key: str, day_key: str
-    ) -> list[dict[str, Any]]:
-        """
-        This method help with filtering data based on the timeframe provided.
-
-        Parameters
-        ----------
-        filters: dict[str, Any]
-            A filter to collect the desired data from OutputManager.
-        date_cutoff: datetime
-            Date before which all data collected from the Output Manager is filtered out.
-        year_key: str
-            How the key is named for the year data retrieved from OM.
-        day_key: str
-            How the key is named for the day data retrieved from OM.
-
-        Returns
-        -------
-        list[dict[str, Any]]
-            Filtered result of the retrieved data.
+        for field_name in harvest_yield_by_field:
+            harvest_dates = sorted(list(harvest_yield_by_field[field_name].keys()))
+            last_harvest_date = -1
+            for harvest_date in harvest_dates:
+                feed_id = harvest_yield_by_field[field_name][harvest_date]["feed_id"]
+                if feed_id is None:
+                    last_harvest_date = harvest_date
+                    continue
+                if feed_id not in total_farmgrown_feed_emission_and_resource_by_feed_id:
+                    total_farmgrown_feed_emission_and_resource_by_feed_id[feed_id] = {
+                        "nitrous_oxide_emissions": 0.0,
+                        "ammonia_emissions": 0.0,
+                        "fertilizer_N": 0.0,
+                        "fertilizer_P": 0.0,
+                        "fertilizer_K": 0.0,
+                        "manure_N": 0.0,
+                    }
+                total_farmgrown_feed_emission_and_resource_by_feed_id[feed_id]["nitrous_oxide_emissions"] += sum(
```

Comment:
I know we talked about it but all these are duplicate code and could be combined - just making a note to make sure it's acknowledged and addressed somewhere down the line.